### PR TITLE
CLOUDSTACK-8944 Template download possible from new nfs secondary storage before download is 100 % complete

### PR DESCRIPTION
**Pull request by yvsubhash**
Steps to test This
1. Add a new NFS storage
2. Start dowloading any public template before it is full dowloaded to new nfs storage
3. The link should always be from any of the older storages where it is fully downloaded

### CODE REVIEW

File: server/src/com/cloud/template/TemplateManagerImpl.java [hunk 1]
```diff
@@ -509,7 +509,7 @@ private String extract(Account caller, Long templateId, String url, Long zoneId,
             for (DataStore store : ssStores) {
                 tmpltStoreRef = _tmplStoreDao.findByStoreTemplate(store.getId(), templateId);
                 if (tmpltStoreRef != null) {
-                    if (tmpltStoreRef.getDownloadState() == com.cloud.storage.VMTemplateStorageResourceAssoc.Status.DOWNLOADED) {
+                    if (tmpltStoreRef.getDownloadState() == com.cloud.storage.VMTemplateStorageResourceAssoc.Status.DOWNLOADED && tmpltStoreRef.getDownloadPercent() == 100) {
```
Comment (sateesh-chodapuneedi) [inline server/src/com/cloud/template/TemplateManagerImpl.java]: @yvsubhash thanks for the patch.

Can we address this (add this check) while setting the download state to DOWNLOADED? That way unless download progress is 100%, the template would not move to DOWNLOADED state.
Also good add a unit here, checking if template download progress is 100 if template state is DOWNLOADED.

**Comment by remibergsma** [comment]
@yvsubhash can you please respond to the comments and also rebase with current master please?

**Comment by rhtyd** [comment]
@yvsubhash please rebase against latest master and push -f, update on status of your PR

tag:easypr

**Comment by bvbharatk** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 96
 Hypervisor xenserver
 NetworkType Advanced
 Passed=73
 Failed=0
 Skipped=3

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**

**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vpc_vpn.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by jayakarteek** [comment]
unable to reproduce in ACS environment

**Comment by yvsubhash** [comment]
Closing this PR, as this is no longer needed

**Pull request closed without merging**